keyform/views.py

```diff
@@ -1,35 +1,65 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
-
+from datetime import datetime, timedelta
 from django.views.generic import FormView, UpdateView, CreateView
 from django.views.generic.list import ListView
 from keyform.forms import CreateForm, RequestFormSet, EditForm
-from keyform.models import Request
+from keyform.models import Request, Building
 from django.urls import reverse_lazy
 from django.http import HttpResponseRedirect
 from django.db.models import Count
 from django.contrib.auth.mixins import LoginRequiredMixin
+from django.utils.dateparse import parse_date
+
 
 class HomeView(LoginRequiredMixin, ListView):
     model = Request
     template_name = "keyform/home.html"
     paginate_by = 25
+    valid_params = ['amt_recieved', 'bpn', 'building__name', 'building_id', 'charge_amount', 'charged_on_rcr',
+                    'comment', 'created_timestamp', 'id', 'keydata__room_number', 'keydata__core_number', 'keydata__key_number',
+                    'payment_method', 'reason_for_request', 'staff', 'staff_id', 'status', 'student_name']
+
+    def get_context_data(self):
+        context = super(HomeView, self).get_context_data()
+        context["request_types"] = Request.REQUEST_TYPES
+        context["status_types"] = Request.STATUS_TYPES
+        context["buildings"] = Building.objects.all()
+        data = self.request.GET.copy()
+        for k, v in self.request.GET.items():
+            if k not in self.valid_params or not v:
+                del data[k]
+        context["data"] = data
+        return context
 
     def get_queryset(self):
-        q_set = super(HomeView, self).get_queryset()
-        q_set = q_set.select_related('building')
-        q_set = q_set.prefetch_related('keydata_set')
-        q_set = q_set.annotate(num_comments=Count('comment'))
-        return q_set
+        qset = super(HomeView, self).get_queryset()
+        qset = qset.select_related('building')
+        qset = qset.prefetch_related('keydata_set')
+        qset = qset.annotate(num_comments=Count('comment'))
+
+        for item, value in self.request.GET.items():
+            if str(item) in self.valid_params:
+                if value != '':
+                    qset = qset.filter(**{item + '__icontains': value})
+
+        converted_start_date = parse_date(self.request.GET.get('start_date', '')) or datetime.min
+        converted_end_date = parse_date(self.request.GET.get('end_date', '')) or datetime.max
```

Comment:
`datetime.max` -> `datetime.max.replace(tzinfo=utc)`
Also you need to import `from django.utils.timezone import utc`